datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -122,42 +136,73 @@ impl<T: CursorValues> Ord for Cursor<T> {
 /// Used for sorting when there are multiple columns in the sort key
 #[derive(Debug)]
 pub struct RowValues {
-    rows: Rows,
+    rows: Arc<Rows>,
+
+    /// Lower bound of windowed RowValues.
+    offset: usize,
+    /// Upper bound of windowed RowValues (not inclusive).
+    limit: usize,
 
     /// Tracks for the memory used by in the `Rows` of this
     /// cursor. Freed on drop
     #[allow(dead_code)]
-    reservation: MemoryReservation,
+    reservation: Arc<MemoryReservation>,
 }
 
 impl RowValues {
-    /// Create a new [`RowValues`] from `rows` and a `reservation`
-    /// that tracks its memory. There must be at least one row
+    /// Create a new [`RowValues`] from `Arc<Rows>`.
```

Comment:
I think it takes a owned `Rows`, and the `Arc` is created internally

```suggestion
    /// Create a new [`RowValues`] from `Rows`.
```